specification/dartLangSpec.tex

```diff
@@ -11677,113 +12041,209 @@ \subsection{Identifier Reference}
   \alt \STATIC{}
   \alt \TYPEDEF{}
 
-<IDENTIFIER\_START> ::= <IDENTIFIER\_START\_NO\_DOLLAR>
-  \alt `$'
+<IDENTIFIER\_START> ::= <IDENTIFIER\_START\_NO\_DOLLAR> | `$'
 
-<IDENTIFIER\_START\_NO\_DOLLAR> ::= <LETTER>
-  \alt `_'
+<IDENTIFIER\_START\_NO\_DOLLAR> ::= <LETTER> | `_'
 
 <IDENTIFIER\_PART\_NO\_DOLLAR> ::= \gnewline{}
-  <IDENTIFIER\_START\_NO\_DOLLAR>
-  \alt <DIGIT>
+  <IDENTIFIER\_START\_NO\_DOLLAR> | <DIGIT>
 
-<IDENTIFIER\_PART> ::= <IDENTIFIER\_START>
-  \alt <DIGIT>
+<IDENTIFIER\_PART> ::= <IDENTIFIER\_START> | <DIGIT>
 
 <qualified> ::= <identifier> (`.' <identifier>)?
 \end{grammar}
 
 \LMHash{}%
 A built-in identifier is one of the identifiers produced by the production
 \synt{BUILT\_IN\_IDENTIFIER}.
-It is a compile-time error if a built-in identifier is used as the declared name of a prefix, class, type parameter or type alias.
-It is a compile-time error to use a built-in identifier other than \DYNAMIC{} in a type annotation or type parameter.
+It is a compile-time error if a built-in identifier is used as
+the declared name of a prefix, class, type parameter or type alias.
+It is a compile-time error to use a built-in identifier
+other than \DYNAMIC{} or \FUNCTION{}
+in a type annotation or a type parameter bound.
 
-\rationale{
-Built-in identifiers are identifiers that are used as keywords in Dart, but are not reserved words in Javascript.
-To minimize incompatibilities when porting Javascript code to Dart, we do not make these into reserved words.
+\rationale{%
+Built-in identifiers are identifiers that are used as keywords in Dart,
+but are not reserved words.
 A built-in identifier may not be used to name a class or type.
 In other words, they are treated as reserved words when used as types.
-This eliminates many confusing situations without causing compatibility problems.
-After all, a Javascript program has no type declarations or annotations so no clash can occur.
-Furthermore, types should begin with an uppercase letter (see the appendix) and so no clash should occur in any Dart user program anyway.
+This eliminates many confusing situations,
+both for human readers and during parsing.%
 }
 
 \LMHash{}%
-It is a compile-time error if either of the identifiers \AWAIT{} or \YIELD{} is used as an identifier in a function body marked with either \ASYNC{}, \code{\ASYNC*} or \code{\SYNC*}.
+It is a compile-time error if either of the identifiers \AWAIT{} or \YIELD{}
+is used as an identifier in a function body
+marked with either \ASYNC{}, \code{\ASYNC*} or \code{\SYNC*}.
```

Comment:
(Nitpick, should affect parameters, but not nested functions.
```dart
foo(int await) async {  // should probably not be allowed
   bar(int await) { print(await); } // Should actually be allowed.
}
```

(I'll stop digressing now).

Reply:
I think it won't make anybody happier to allow that nested case. I'd prefer to leave it as it is. ;-)

Reply:
I'll mark this as resolved; please create an issue in order to get the enhancement where `await` is allowed in such places where it is currently an error.